openhcl/hcl/src/ioctl/x64.rs

```diff
@@ -403,4 +403,56 @@ impl<'a> BackingPrivate<'a> for MshvX64<'a> {
 
         Ok(None)
     }
+
+    fn flush_register_page(runner: &mut ProcessorRunner<'a, Self>) {
+        let Some(reg_page) = runner.reg_page_mut() else {
+            return;
+        };
+
+        // Collect any dirty registers.
+        let mut regs: Vec<(HvX64RegisterName, HvRegisterValue)> = Vec::new();
+        if (reg_page.dirty & (1 << hvdef::HV_X64_REGISTER_CLASS_IP)) != 0 {
+            regs.push((HvX64RegisterName::Rip, reg_page.rip.into()));
+        }
+        if (reg_page.dirty & (1 << hvdef::HV_X64_REGISTER_CLASS_GENERAL)) != 0 {
+            regs.push((
+                HvX64RegisterName::Rsp,
+                reg_page.gp_registers
+                    [(HvX64RegisterName::Rsp.0 - HvX64RegisterName::Rax.0) as usize]
+                    .into(),
+            ));
+        }
+        if (reg_page.dirty & (1 << hvdef::HV_X64_REGISTER_CLASS_FLAGS)) != 0 {
+            regs.push((HvX64RegisterName::Rflags, reg_page.rflags.into()));
+        }
+        if (reg_page.dirty & (1 << hvdef::HV_X64_REGISTER_CLASS_SEGMENT)) != 0 {
+            let segment_regs = reg_page
+                .segment
+                .iter()
+                .copied()
+                .enumerate()
+                .map(|(i, val)| {
+                    (
+                        HvX64RegisterName::from(HvRegisterName(HvX64RegisterName::Es.0 + i as u32)),
+                        HvRegisterValue::from(val),
+                    )
+                })
+                .collect::<Vec<_>>();
+            regs.extend_from_slice(segment_regs.as_slice());
```

Comment:
Avoid an extra allocation. Just do `regs.extend(segment_regs)`, and remove the `.collect` on `segment_regs`.